src/Compilers/CSharp/Portable/Binder/ExecutableCodeBinder.cs

```diff
@@ -113,7 +113,7 @@ public static void ValidateIteratorMethod(CSharpCompilation compilation, MethodS
                 {
                     diagnostics.Add(ErrorCode.ERR_BadIteratorArgType, parameter.GetFirstLocation());
                 }
-                else if (parameter.Type.IsUnsafe())
+                else if (parameter.Type.IsPointerOrFunctionPointer())
                 {
                     diagnostics.Add(ErrorCode.ERR_UnsafeIteratorArgType, parameter.GetFirstLocation());
```

Comment:
>diagnostics.Add(ErrorCode.ERR_UnsafeIteratorArgType, parameter.GetFirstLocation()); [](http://example.com/codeflow?start=20&length=83)

It looks like this error mentions return type as well, but I do not see where we check the return type. Are we missing the check, or the wording is wrong?

Reply:
This question is relevant if we decide to adjust the wording of the message.

Reply:
For iterators, yielding a pointer type is indeed forbidden (pointer types cannot be used as type arguments, guarded with normal "error CS0306: The type '...' may not be used as a type argument").
I agree that this error message doesn't need to mention yielded types, but it also seems fine to leave as-is (the message is not wrong) and it's tangential to this PR.